NDTensors/src/diag/tensoralgebra/contract.jl

```diff
@@ -58,6 +58,29 @@ function _contract!!(
   return R
 end
 
+function contract!(
+  output_tensor::Exposed{<:AbstractArray,<:DiagTensor},
+  labelsoutput_tensor,
+  tensor1::Exposed,
+  labelstensor1,
+  tensor2::Exposed,
+  labelstensor2,
+  α::Number=one(Bool),
+  β::Number=zero(Bool),
+)
+  elt = eltype(unexpose(output_tensor))
+  @assert α ≈ one(elt)
+  @assert β ≈ zero(elt)
```

Comment:
```suggestion
  @assert α == one(Bool)
  @assert β == zero(Bool)
```
Was there an issue with using strict equality instead of approximate equality? It seems better to be strict here.

Reply:
Also for the sake of comparison the element type of the number shouldn't matter.